src/backend/distributed/executor/local_executor.c

```diff
@@ -220,9 +220,15 @@ ExecuteLocalTaskList(CitusScanState *scanState, List *taskList)
 
 		LogLocalCommand(task);
 
-		char *shardQueryString = task->queryStringLazy
-								 ? task->queryStringLazy
-								 : "<optimized out by local execution>";
+		char *shardQueryString = NULL;
+		if (GetTaskQueryType(task) == TASK_QUERY_TEXT)
+		{
+			shardQueryString = TaskQueryStringForAllPlacements(task);
+		}
+		else
+		{
+			shardQueryString = "<optimized out by local execution>";
```

Comment:
could use a comment, e.g. `/* avoid the overhead of deparsing when using local execution */`